### PR TITLE
doc: Add top-level navigation grid

### PR DESCRIPTION
**Pull request by nashif**
Make the documentation page a little bit more interesting with top-level
navigation grid highlighting the most important sections with a brief
description.

### CODE REVIEW

**Comment by nashif** [comment]
preview: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/12357/docs/index.html

**Comment by dbkinder** [comment]
I see what your doing... looks better, but harder to maintain, and I doubt the PDF version will be happy (though I've seen comments that we might not be caring that much about generating a PDF of the complete doc site).  Let me look into how we can do something similar without diving into raw HTML.

**Comment by nashif** [comment]
> I see what your doing... looks better, but harder to maintain

For the few  top level links I think that is something we can live with.
 
> and I doubt the PDF version will be happy

should work just fine with the PDF version, I tested that.

> Let me look into how we can do something similar without diving into raw HTML.

ok

**Comment by nashif** [comment]
updated with some icons, pleaseholders only, hence the DNM

**Comment by dbkinder** [comment]
I'm not having much luck doing this visual change with just reST (without resorting to raw html as you've done), so I'm OK with this change, except for the "Fork me in GitHub" corner label on every page.  That we can do without.

File: doc/_templates/layout.html [hunk 1]
```diff
@@ -7,4 +7,5 @@
     </div>
   {% endif %}
   {{ super() }}
+<a href="https://github.com/zephyrproject-rtos/zephyr"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png" alt="Fork me on GitHub"></a>
```
Comment (dbkinder) [inline doc/_templates/layout.html]: I don't like this corner banner on every page.  Seems distracting and unnecessary.

**Review by dbkinder — DISMISSED** [review]
Except for the corner banner, this looks OK.

File: doc/_templates/layout.html (hunk 1 above)
Comment (nashif) [reply doc/_templates/layout.html]: we can have it on the top page only if that helps.

**Review by dbkinder — APPROVED** [review]
OK, thanks.  You got rid of all the pictures (except for the kite)?  I can find us some new ones if you'd like and add them later...

**Comment by nashif** [comment]
> I can find us some new ones if you'd like and add them later...

yes please, those I had there were ugly and were added just as an example. If you find nice ones following same design pattern that would be great.

**Pull request merged**